config/routes.rb

```diff
@@ -20,21 +20,25 @@
       get '/email', to: 'email#show'
       post '/email', to: 'email#create'
 
-      constraints(-> { Setting.signal_server_phone_number.present? }) do
+      constraints(-> { Setting.signal_configured? }) do
         get '/signal/', to: 'signal#show'
         get '/signal/link/', to: 'signal#link', as: 'signal_link'
         post '/signal/', to: 'signal#create'
       end
 
-      get '/threema/', to: 'threema#show'
-      post '/threema/', to: 'threema#create'
+      constraints(-> { Setting.threema_configured? }) do
+        get '/threema/', to: 'threema#show'
+        post '/threema/', to: 'threema#create'
+      end
 
-      get '/telegram/', to: 'telegram#show'
-      get '/telegram/link/:telegram_onboarding_token', to: 'telegram#link', as: 'telegram_link'
-      get '/telegram/fallback/:telegram_onboarding_token', to: 'telegram#fallback', as: 'telegram_fallback'
-      post '/telegram/', to: 'telegram#create'
+      constraints(-> { Setting.telegram_configured? }) do
+        get '/telegram/', to: 'telegram#show'
+        get '/telegram/link/:telegram_onboarding_token', to: 'telegram#link', as: 'telegram_link'
+        get '/telegram/fallback/:telegram_onboarding_token', to: 'telegram#fallback', as: 'telegram_fallback'
+        post '/telegram/', to: 'telegram#create'
+      end
```

Comment:
This is partially clean up. Now that we have these class methods, we want to use them to not create routes for channels that are not configured.

Reply:
Are those methods for the onboarding only? If so shouldn't they check for allow_onboarding?

Reply:
Yes, they are for onboarding. We filter out the available channels here https://github.com/tactilenews/100eyes/pull/1818/files#diff-c013d1db563a45b6fcd49779b9cade0a0748018246f3f8f919a7c87ac4f39c34R16, but if the route is available for a configured channel, like email(which needs to be configured for user managment), and the client wants to disallow onboarding, someone who knows what they are doing would still be able to onboard. So, good call, I'll updated these routes.

Reply:
Ok, I've updated the constraints with the additional condition that onboarding must be allowed. I've also noticed that I missed that there were no constraints on the email onboarding paths, which is the reason we added the feature to disallow onboarding in the first place. I've added the constraint now.

See 4e87091a

I've updated the routing spec we have for WhatsApp. We don't have any other routing specs. I have seen a lot of posts discouraging the use of them, not sure your opinion. This also brought to my attention that we do not have any WhatsApp request specs for onboarding besides these limited routing specs. I've added an issue for it https://github.com/tactilenews/100eyes/issues/1848

I've also added request specs to check that the constraints are properly set up. See c825f9fe

Reply:
Well just my opinion if I understand it correctly: I think you are already answering yourself to some extent in that comment. I think it's absolutely fine to test for that 404s if that is part of the application logic. Regarding the positive cases it would be better to have request specs instead of routing specs that only test for a 200 - but the later might be better than nothing. So I'm fine with a follow up